pkg/apis/certmanager/validation/certificate_test.go

```diff
@@ -178,6 +178,104 @@ func TestValidateCertificate(t *testing.T) {
 				field.Required(fldPath.Child("acme", "config"), "no ACME solver configuration specified for domain \"commonname\""),
 			},
 		},
+		"valid certificate with rsa keyAlgorithm specified and no keySize": {
+			cfg: &v1alpha1.Certificate{
+				Spec: v1alpha1.CertificateSpec{
+					CommonName:   "testcn",
+					SecretName:   "abc",
+					IssuerRef:    validIssuerRef,
+					KeyAlgorithm: v1alpha1.RSAKeyAlgorithm,
+				},
+			},
+		},
+		"valid certificate with ecdsa keyAlgorithm specified and no keySize": {
+			cfg: &v1alpha1.Certificate{
+				Spec: v1alpha1.CertificateSpec{
+					CommonName:   "testcn",
+					SecretName:   "abc",
+					IssuerRef:    validIssuerRef,
+					KeyAlgorithm: v1alpha1.ECDSAKeyAlgorithm,
+				},
+			},
+		},
+		"valid certificate with keyAlgorithm not specified and keySize specified": {
+			cfg: &v1alpha1.Certificate{
+				Spec: v1alpha1.CertificateSpec{
+					CommonName: "testcn",
+					SecretName: "abc",
+					IssuerRef:  validIssuerRef,
+					KeySize:    2048,
+				},
+			},
+		},
```

Comment:
Can we add a few more valid success cases? (e.g. for ecdsa with a valid size, explicit rsa with a valid size)

Reply:
Added